Zend/zend_alloc.c

```diff
@@ -663,35 +696,47 @@ static zend_always_inline int zend_mm_bitset_is_free_range(zend_mm_bitset *bitse
 
 static void *zend_mm_chunk_alloc_int(size_t size, size_t alignment)
 {
-	void *ptr = zend_mm_mmap(size);
-
+#ifdef _WIN32
+	void *ptr = NULL;
+	size_t offset;
+#ifndef _WIN64
+	/* 32-bit build has limited memory address space and partical deallocation is not supported,
+	 * try to reserve exact aligned size first, GH-9650 */
+	ptr = zend_mm_mmap_reserve(size + alignment);
 	if (ptr == NULL) {
 		return NULL;
-	} else if (ZEND_MM_ALIGNED_OFFSET(ptr, alignment) == 0) {
-#ifdef MADV_HUGEPAGE
-		if (zend_mm_use_huge_pages) {
-			madvise(ptr, size, MADV_HUGEPAGE);
-		}
-#endif
-		return ptr;
-	} else {
-		size_t offset;
+	}
 
-		/* chunk has to be aligned */
-		zend_mm_munmap(ptr, size);
-		ptr = zend_mm_mmap(size + alignment - REAL_PAGE_SIZE);
-#ifdef _WIN32
-		offset = ZEND_MM_ALIGNED_OFFSET(ptr, alignment);
-		zend_mm_munmap(ptr, size + alignment - REAL_PAGE_SIZE);
-		ptr = zend_mm_mmap_fixed((void*)((char*)ptr + (alignment - offset)), size);
-		offset = ZEND_MM_ALIGNED_OFFSET(ptr, alignment);
-		if (offset != 0) {
-			zend_mm_munmap(ptr, size);
+	offset = ZEND_MM_ALIGNED_OFFSET(ptr, alignment);
+	if (offset != 0) {
+		offset = alignment - offset;
+	}
+	zend_mm_munmap(ptr, size + alignment);
+	ptr = zend_mm_mmap_reserve_try((void*)((char*)ptr + offset), size);
+#endif
```

Comment:
Remove redundant ifndef _WIN64 check?

Reply:
`_WIN64` is detecting x64 only (`_WIN86` is however for both, probably by some BC reasons Microsoft decised to do so...)